packages/code-studio/src/main/AppInit.tsx

```diff
@@ -167,216 +95,159 @@ function AppInit(props: AppInitProps) {
+          const platformOverrides = isMac
+            ? shortcutOverrides.mac ?? {}
+            : shortcutOverrides.windows ?? {};
+
+          Object.entries(platformOverrides).forEach(([id, keyState]) => {
+            ShortcutRegistry.get(id)?.setKeyState(keyState);
+          });
+
+          const dashboardData = {
+            filterSets: data.filterSets,
+            links: data.links,
+          };
+
+          const configs = await client.getServerConfigValues();
+          const serverConfig = new Map(configs);
+
+          const user: User = {
+            name,
+            operateAs: name,
+            groups: [],
+            permissions: {
+              isACLEditor: false,
+              isSuperUser: false,
+              isQueryViewOnly: false,
+              isNonInteractive: false,
+              canUsePanels: true,
+              canCreateDashboard: true,
+              canCreateCodeStudio: true,
+              canCreateQueryMonitor: true,
+              canCopy: !(
+                serverConfig.get('internal.webClient.appInit.canCopy') ===
+                'false'
+              ),
+              canDownloadCsv: !(
+                serverConfig.get(
+                  'internal.webClient.appInit.canDownloadCsv'
+                ) === 'false'
+              ),
+            },
+          };
+
+          setActiveTool(ToolType.DEFAULT);
+          setServerConfigValues(serverConfig);
+          setCommandHistoryStorage(commandHistoryStorage);
+          setDashboardData(DEFAULT_DASHBOARD_ID, dashboardData);
+          setFileStorage(fileStorage);
+          setLayoutStorage(layoutStorage);
+          setDashboardConnection(DEFAULT_DASHBOARD_ID, connection);
+          if (sessionWrapper !== undefined) {
+            setDashboardSessionWrapper(DEFAULT_DASHBOARD_ID, sessionWrapper);
+          }
+          setUser(user);
+          setWorkspaceStorage(workspaceStorage);
+          setWorkspace(loadedWorkspace);
+        } catch (e) {
+          log.error(e);
+          setError(e);
+        }
+      }
+      loadApp();
     },
-    [initClient, initFonts]
+    [
+      client,
+      connection,
+      setActiveTool,
+      setCommandHistoryStorage,
+      setDashboardData,
+      setFileStorage,
+      setLayoutStorage,
+      setDashboardConnection,
+      setDashboardSessionWrapper,
+      setUser,
+      setWorkspace,
+      setWorkspaceStorage,
+      setServerConfigValues,
+    ]
   );
 
-  const isLoading = (workspace == null && error == null) || isFontLoading;
+  const isLoading = workspace == null && error == null;
   const isLoaded = !isLoading && error == null;
-  const errorMessage =
-    error != null || disconnectError != null
-      ? `${error ?? disconnectError}`
-      : null;
+  const errorMessage = error != null ? `${error}` : null;
 
   return (
     <>
       {isLoaded && <App />}
       <LoadingOverlay
-        isLoading={isLoading}
+        isLoading={isLoading && errorMessage == null}
```

Comment:
Why do we also need to check the error message here? Seems like if that's the case then `LoadingOverlay` should handle the case where `isLoading === true` and `errorMessage != null`

Reply:
Yea I didn't want to change that behaviour... looking at `LoadingOverlay`, you can display a loading spinner along with a message underneath, though it is poorly named for that case.